Add more SCE contents to merged objects page (#237)
AlexsLemonade/scpca-docs: docs/merged_objects.md

| `singler_celltype_annotation`  | If cell typing with `SingleR` was performed, the annotated cell type. Cells labeled as `NA` are those which `SingleR` could not confidently annotate. If cell typing was performed for some but not all libraries in the merged object, libraries without annotations will be labeled `"Cell type annotation not performed"` |
| `singler_celltype_ontology`  | If cell typing with `SingleR` was performed with ontology labels, the annotated cell type's ontology ID. Cells labeled as `NA` are those which `SingleR` could not confidently annotate. If cell typing was performed for some but not all libraries in the merged object, libraries without annotations will be labeled `"Cell type annotation not performed"` |
| `cellassign_celltype_annotation`  | If cell typing with `CellAssign` was performed, the annotated cell type. Cells labeled as `"other"` are those which `CellAssign` could not confidently annotate. If cell typing was performed for some but not all libraries in the merged object, libraries without annotations will be labeled `"Cell type annotation not performed"`  |

> | `cellassign_max_prediction`  | If cell typing with `CellAssign` was performed, the annotation's prediction score (probability)  |

Author: argh foiled by copy/paste 😬

> | `cellassign_max_prediction`  | If cell typing with `CellAssign` was performed, the annotation's prediction score (probability)  |

Author: Ok, I've gotten all the `colData` here: https://github.com/AlexsLemonade/scpca-docs/pull/237/commits/7c7861eb87e71cd94b4cc9be410a02b86d3ece3e

And found something else we need to fix! Looks like merged objects are missing `scpca_filter` and `adt_scpca_filter` (if at least 1 library ahs cite). I feel pretty sure we want to keep those columns around! I'll file a PR in `scpca-nf` to retain those columns.

> | `cellassign_max_prediction`  | If cell typing with `CellAssign` was performed, the annotation's prediction score (probability)  |

Author: Note, currently updating colData bits to properly handle multiplex which won't have this (related to https://github.com/AlexsLemonade/scpca-docs/pull/237#discussion_r1457526408)

| ------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `library_id`       | A vector of library IDs which are included in the merged object, each in the form `SCPCL000000`                                                                                                                                                                                             |
| `sample_id`        | A vector of sample IDs which are included in the merged object, each in the form  `SCPCS000000`. If there are multiple samples in this library, sample IDs will be given as a comma-separated list                                                                                           |
| `library_metadata` | A named list of lists, for each library ID. Each list contains the metadata fields for the given library `{ref}as they would appear in an individual library object<sce_file_contents:singlecellexperiment sample metadata>` |

> | `merged_highly_variable_genes`      | A vector of highly variable genes used for performing dimensionality reduction on the merged object, determined using `scran::modelGeneVar`, specifying each library as a separate block, and `scran::getTopHVGs`    |

Reviewer: I think I would add a note here too that any projects with multiplexed libraries will contain an additional `sample_metadata` data frame and link to the section below.

> | `merged_highly_variable_genes`      | A vector of highly variable genes used for performing dimensionality reduction on the merged object, determined using `scran::modelGeneVar`, specifying each library as a separate block, and `scran::getTopHVGs`    |

Author: I ended up actually adding a row to the table, is that ok?

There is also a `spliced` assay which contains the counts matrix with only reads from spliced cDNA.

### SingleCellExperiment cell metrics

Cell metrics calculated from the RNA-seq expression data are stored as a `DataFrame` in the `colData` slot, where row names are the cell barcode prefixed with the originating library id, e.g. `SCPCL000000-{barcode}`.

> This `DataFrame` also contains additional sample metadata information stored in the `colData` slot `DataFrame`, as described in [`Sample metadata` section](#singlecellexperiment-sample-metadata), for non-multiplexed samples.

Reviewer: ```suggestion
This `DataFrame` also contains additional sample metadata information stored in the `colData` slot `DataFrame` for all projects that do not contain multiplexed libraries.
Read more about the sample metadata that is included in the [`Sample metadata` section](#singlecellexperiment-sample-metadata), 
```
Since the presence of sample metadata is dependent on any libraries in a project being multiplexed, let's specify that.

| `cellassign_max_prediction`      | If cell typing with `CellAssign` was performed, the annotation's prediction score (probability)                                                                                                                                                                                                                                                                 |

Unlike for {ref}`individual SCE objects<sce_file_contents:singlecellexperiment cell metrics`, cluster assignments are not included in the `colData`.

### SingleCellExperiment gene information and metrics

Gene information and metrics calculated from the RNA-seq expression data are stored as a `DataFrame` in the `rowData` slot, with the Ensembl ID as the names of the rows.

```r
rowData(merged_sce) # gene metrics
```

The following columns are included for all genes.
The columns `mean` and `detected` will appear for each library id included in the merged object, named as shown in the table below.

> However, there will only be a single `gene_symbol` column, as this information equally pertains to all libraries.

Reviewer: ```suggestion
However, there will only be a single `gene_symbol` and `gene_id` column, as this information equally pertains to all libraries.
```
Is it `gene_id` or `gene_ids`, I think `gene_id`?

Author: Ah, `gene_ids`! I'll add this on my end.

```
> sce <- readRDS("SCPCL000001_processed.rds")
> names(rowData(sce))
[1] "gene_ids"    "gene_symbol" "mean"        "detected"  
```

| `usa_mode`          | Boolean indicating whether quantification was done using `alevin-fry` USA mode                                                 |
| `af_num_cells`      | Number of cells reported by `alevin-fry`                                                                                       |
| `tech_version`      | A string indicating the technology and version used for the single-cell library, such as 10Xv2, 10Xv3, or 10Xv3.1              |
| `assay_ontology_term_id` | A string indicating the [Experimental Factor Ontology](https://www.ebi.ac.uk/ols/ontologies/efo) term id associated with the `tech_version`  |
| `seq_unit`         | `cell` for single-cell samples or `nucleus` for single-nucleus samples                                                          |
| `transcript_type`   | Transcripts included in gene counts: `spliced` for single-cell samples and `unspliced` for single-nuclei                       |
| `sample_metadata`   | Data frame containing metadata for each sample included in the library (see the [`Sample metadata` section below](#singlecellexperiment-sample-metadata)) |

> | `miQC_model`        | The model object that `miQC` fit to the data and was used to calculate `prob_compromised`. Only present for `filtered` and `processed` objects |

Reviewer: I would remove every reference to `Only present for x objects`. There's only one type of merged object.

Author: 🤦‍♀️

| `cellassign_reference_organs` | If cell typing with `CellAssign` was performed, a comma-separated list of organs and/or tissue compartments from which marker genes were obtained to create the reference. Only present for `processed` objects |
| `cellassign_reference_source`  | If cell typing with `CellAssign` was performed, the source of the reference dataset (default is [`PanglaoDB`](https://panglaodb.se/)). Only present for `processed` objects |
| `cellassign_reference_version`  | If cell typing with `CellAssign` was performed, the version of the reference dataset source. For references obtained from `PanglaoDB`, the version scheme is a date in ISO8601 format. Only present for `processed` objects |

Unlike for {ref}`individual SingleCellExperiment objects<sce_file_contents:singlecellexperiment sample metadata`, cluster algorithm parameters are not included in these metadata lists because clusters themselves are not included in the merged object.

### SingleCellExperiment sample metadata

Sample metadata describing each sample included in the merged object is stored in one of two locations, depending on the sample type:

> If the project does not contain multiplexed samples, this information is stored in the `colData` slot's `DataFrame`, along with [cell metrics](#singlecellexperiment-cell-metrics).

Reviewer: ```suggestion
If the project does_not_ contain multiplexed libraries, this information can be found as additional columns in the `colData` slot's `DataFrame`, along with [cell metrics](#singlecellexperiment-cell-metrics).
```

```r
colData(merged_sce) # sample metadata for projects without multiplexing
```

> If the project contains is multiplexed samples, this information is stored in the `metadata` slot in the field `sample_metadata` as a `data.frame`:

Reviewer: ```suggestion
If the project contains multiplexed libraries, this information is stored in the `metadata` slot in the `sample_metadata` field as a `data.frame`:
```

```r
metadata(merged_sce)$sample_metadata # sample metadata for proejcts with multiplexed samples
```

> Note that, in `{ref}in individual library objects<sce_file_contents:singlecellexperiment sample metadata>`, this sample metadata information is instead stored in the `SingleCellExperiment` object's `metadata` slot.

Reviewer: ```suggestion
Note that, in {ref} `individual library objects<sce_file_contents:singlecellexperiment sample metadata>`, this sample metadata information is instead stored in the `SingleCellExperiment` object's `metadata` slot.
```

There is also a `spliced` assay which contains the counts matrix with only reads from spliced cDNA.

### SingleCellExperiment cell metrics

Cell metrics calculated from the RNA-seq expression data are stored as a `DataFrame` in the `colData` slot, where row names are the cell barcode prefixed with the originating library id, e.g. `SCPCL000000-{barcode}`.

> This `DataFrame` also contains additional sample metadata information stored in the `colData` slot `DataFrame` for all projects that do not contain multiplexed libraries.

Reviewer: ```suggestion
This `DataFrame` also contains additional sample metadata information stored in the `colData` slot for all projects that do not contain multiplexed libraries.
```
I don't think we need the extra `DataFrame`.

```r
metadata(merged_sce) # experiment metadata
```

| Item name                      | Contents                                                                                                                                                                                                                                                                                                                                               |
| ------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `library_id`                   | A vector of library IDs which are included in the merged object, each in the form `SCPCL000000`                                                                                                                                                                                                                                                        |

> | `sample_id`                    | A vector of sample IDs which are included in the merged object, each in the form  `SCPCS000000`. If there are multiple samples in this library, sample IDs will be given as a comma-separated list                                                                                                                                                     |

Reviewer: ```suggestion
| `sample_id`                    | A vector of sample IDs which are included in the merged object, each in the form  `SCPCS000000`.
```

| `library_metadata`             | A list of the library metadata for each library. Each list is named with the appropriate library ID and contains the library metadata fields for the given library {ref}`as they would appear in an individual library object<sce_file_contents:singlecellexperiment experiment metadata>`. See the table below for a full description of its contents |
| `merged_highly_variable_genes` | A vector of highly variable genes used for performing dimensionality reduction on the merged object, determined using `scran::modelGeneVar`, specifying each library as a separate block, and `scran::getTopHVGs`                                                                                                                                      |

> | `sample_metadata`              | A data frame of additional sample metadata, as described in the [`Sample metadata` section below](#singlecellexperiment-sample-metadata). Only present for projects with multiplexed libraries                                                                                                                                                                                                                                                                                                                                                       |

Reviewer: Maybe switch this order? I believe sample metadata should be first.

| `cellassign_reference_version`  | If cell typing with `CellAssign` was performed, the version of the reference dataset source. For references obtained from `PanglaoDB`, the version scheme is a date in ISO8601 format |

Unlike for {ref}`individual SingleCellExperiment objects<sce_file_contents:singlecellexperiment sample metadata`, cluster algorithm parameters are not included in these metadata lists because clusters themselves are not included in the merged object.

### SingleCellExperiment sample metadata

Sample metadata describing each sample included in the merged object is stored in one of two locations, depending on the sample type:

If the project does_not_ contain multiplexed libraries, this information can be found as additional columns in the `colData` slot's `DataFrame`, along with [cell metrics](#singlecellexperiment-cell-metrics).

```r
colData(merged_sce) # sample metadata for projects without multiplexing
```

If the project contains multiplexed libraries, this information is stored in the `metadata` slot in the `sample_metadata` field as a `data.frame`:

```r
metadata(merged_sce)$sample_metadata # sample metadata for proejcts with multiplexed samples
```

> Note that, in {ref} `individual library objects<sce_file_contents:singlecellexperiment sample metadata>`, this sample metadata information is instead stored in the `SingleCellExperiment` object's `metadata` slot.

Reviewer: I think I would move this sentence to be right after line 189 and state that just like merged objects with multiplexed libraries, all individual objects will contain the `sample_metadata` in the `metadata` slot.
